lib/search_files.js

```diff
@@ -0,0 +1,291 @@
+var fs = require('fs');
+var os = require('os');
+var path = require('path');
+var glob = require('glob');
+
+
+function isWindows() {
+  return process.platform === 'win32';
+}
+
+function isFunction(v) {
+  return (typeof v === 'function');
+}
+
+function isObject(v) {
```

Comment:
As noted in some other comments, I'd like to use the _.mapValues method too, instead of implementing our own (eachProp).  I'm pretty sure this functionality is going to add much more than the 100ms you mentioned because it has lots of filesystem traversing to do.